x/zkauth/keeper/keeper.go

```diff
@@ -1,34 +1,120 @@
 package keeper
 
 import (
+	"encoding/json"
 	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
 
 	"github.com/Finschia/finschia-sdk/codec"
 	storetypes "github.com/Finschia/finschia-sdk/store/types"
-	sdktypes "github.com/Finschia/finschia-sdk/types"
+	sdk "github.com/Finschia/finschia-sdk/types"
 	"github.com/Finschia/finschia-sdk/x/zkauth/types"
 	"github.com/Finschia/ostracon/libs/log"
 )
 
+const JwkFileName = "jwk.json"
+
 type Keeper struct {
 	cdc      codec.BinaryCodec
 	storeKey storetypes.StoreKey
-	memKey   storetypes.StoreKey
 }
 
 func NewKeeper(
 	cdc codec.BinaryCodec,
-	storeKey,
-	memKey storetypes.StoreKey,
+	storeKey storetypes.StoreKey,
 ) *Keeper {
 
 	return &Keeper{
 		cdc:      cdc,
 		storeKey: storeKey,
-		memKey:   memKey,
 	}
 }
 
-func (k Keeper) Logger(ctx sdktypes.Context) log.Logger {
+func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
 }
+
+func (k Keeper) FetchJWK(ctx sdk.Context, nodeHome string) {
+	logger := k.Logger(ctx)
+	var defaultZKAuthOAuthProviders = [1]types.OidcProvider{types.Google}
+	var fetchIntervals uint64
+	go func() {
+		for {
+			select {
+			case <-ctx.Context().Done():
+				return
+			default:
+				for _, name := range defaultZKAuthOAuthProviders {
+					provider := types.GetConfig(name)
+					fetchIntervals = provider.FetchIntervals
+
+					resp, err := k.GetJWK(provider.JwkEndpoint)
+					if err != nil {
+						time.Sleep(time.Duration(fetchIntervals) * time.Second)
+						logger.Error(fmt.Sprintf("%s", err))
+						continue
+					}
+
+					file, err := os.Create(filepath.Join(nodeHome, JwkFileName))
+					if err != nil {
+						time.Sleep(time.Duration(fetchIntervals) * time.Second)
+						logger.Error(fmt.Sprintf("%s", err))
+						fmt.Println(err)
```

Comment:
It seems unnecessary because it is output with logger.Error.

Reply:
I forgot to remove it, thanks!